pkg/build/builder/sti.go

```diff
@@ -92,9 +92,16 @@ func (s *S2IBuilder) Build() error {
 	if s.build.Spec.Strategy.SourceStrategy == nil {
 		return fmt.Errorf("the source to image builder must be used with the source strategy")
 	}
-
+	var runtimeImageName string
 	var push bool
 
+	runtimeImage := s.build.Spec.Strategy.SourceStrategy.RuntimeImage
```

Comment:
@PI-Victor 
the From reference is required in a SourceBuildStrategy and Custom strategy, it is optional in a Docker build strategy (you can just have a Dockerfile that contains a FROM line).

we use pointers when things are optional so the value can be left nil.